lib/content-negotiation.js

```diff
@@ -73,11 +84,16 @@ function canUnmarshall(type) {
     return !!unmarshaller(type);
 }
 
-function unmarshaller(type) {
-    const { unmarshall } = mediaTypeMarshallers[type] || {};
+function unmarshaller(mediaType) {
+    const { unmarshall } = mediaTypeMarshallers[mediaType] || {};
     return unmarshall;
 }
 
+function parseContentType(contentType) {
+    const { type, parameters } = contentTypeParser.parse(contentType);
+    return { type, charset: parameters.charset };
+}
+
 function determineContentTypes(contentType, accept) {
     // TODO correctly handle content-type charset, instead of ignoring it
     contentType = (contentType || "text/plain").split(";")[0].trim();
```

Comment:
Not the subject of this PR, but this seems wrong. Default content type should not be text/plain. (not to mention fragile parsing that can now be replaced by `contentTypeParser`, no?

Reply:
Line 84 seems wrong to me as well. `accept` should not default to incoming `contentType`

Reply:
that PR will fix this too.

Reply:
see https://github.com/projectriff/node-function-invoker/pull/179#issuecomment-610481322

Reply:
Changed my mind, will open a subsequent PR for that.

Reply:
Specific issue: https://github.com/projectriff/node-function-invoker/issues/180.